### ISSUE TITLE
Input URL fields don't reserve space for the spinner icon

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58679]
### Description

Noticed while working on https://github.com/WordPress/gutenberg/pull/58505

See also the reunification of these UIs proposed in https://github.com/WordPress/gutenberg/issues/50893

Both the `LinkControl` input field and the URL input field for images don't reserve space for their spinner icon. As such, long text inserted in the input field is partially hidden by the spinner icon.

Generally, I'm not sure that visually placing extraneous control within an input field is ideal in the first place. The visually appearance os such a placement may look more pleasant but that's subjective and not that relevant.

The expectation is to have controls that are visually separated and can be distinguished easily, This applies to buttons and also to 'spinner' icons.

The fact that the spinner icon hides content, although it only does that temporarily, isn't ideal and should be prevented.

Screenshots:

![spinner 01](https://github.com/WordPress/gutenberg/assets/1682452/3e6db28c-480a-4dd3-bccc-97262f993fbe)

![spinner 02](https://github.com/WordPress/gutenberg/assets/1682452/9d291570-5c08-4b4e-979a-25004e48a830)

### Step-by-step reproduction instructions

- Select some text and start inserting a link.
- In the URL input field type some long text and wait to see the spinner icon.
- Observe the spinner icon hides the text you entered.
- Add an Image block and start adding a link to the image.
- In the URL input field type some long text and wait to see the spinner icon.
- Observe the spinner icon hides the text you entered.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
